dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/AbstractInterfaceConfig.java

```diff
@@ -685,7 +685,6 @@ public ApplicationConfig getApplication() {
 
     public void setApplication(ApplicationConfig application) {
         this.application = application;
-        ConfigManager.getInstance().setApplication(this.application);
```

Comment:
For the long-term run, I totally agree we should remove this.
But, before we can provide a fluent style API, as you pointed, sth works like this `DubboConfigBuilder().getInstance().buildApplication(appConfig).buildService(service1, service2)`, I think we should keep it. 
Keep it to solve a problem: guarantee every ServiceConfig/ReferenceConfig uses the same global Config when using API, you can find Spring guarantees this in the current version.